pkg/filterengine/filters/object_annotation_checker.go

```diff
@@ -82,6 +107,101 @@ func isObjectNotifDisabled(obj metaV1.ObjectMeta) bool {
 	return false
 }
 
+// isObjectNamespaceNotifEnabled checks annotation botkube.io/notification on the namespace
+// annotation botkube.io/notification enable/disable all object's event notification in the namespace
+func isObjectNamespaceNotifEnabled(obj metaV1.ObjectMeta) bool {
+
+	namespace := obj.Namespace
+	nsObjectNotifEnabled, found := NamespaceCache.Get(namespace)
+
+	if !found {
+		log.Warnf("Cache Miss!! Namespace '%s' not found in the cache!", namespace)
+
+		// get namespace object
+		nsObject, err := utils.DynamicKubeClient.Resource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "namespaces"}).Get(context.TODO(), namespace, v1.GetOptions{})
+
+		if err != nil {
+			log.Error(err)
+			nsObjectNotifEnabled = "false"
+		} else {
+			annotationsList := nsObject.GetAnnotations()
+			val := annotationsList[NotificationAnnotation]
+			if val != "false" {
+				val = "true"
+			}
+
+			NamespaceCache.Set(nsObject.GetName(), val, cache.NoExpiration)
+			nsObjectNotifEnabled = val
+			log.Debugf("Namespace Cache: %v", NamespaceCache.Items())
+		}
+
+	} else {
+		log.Infof("Cache Hit!! Namespace '%s' found in the cache!", namespace)
+	}
+
+	b1, _ := strconv.ParseBool(nsObjectNotifEnabled.(string))
+	return b1
+
+}
+
+// Registers watch for namespace's events
+func RegisterWatchNamespace() error {
+
+	log.Info("Registering namespace watch")
+
+	// register watch on namespaces
+	nsWatcher, err := utils.DynamicKubeClient.Resource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "namespaces"}).Watch(context.TODO(), v1.ListOptions{})
```

Comment:
The watcher gets closed after some time, please use RetryWatcher - https://pkg.go.dev/k8s.io/client-go/tools/watch#NewRetryWatcher